hikari/audit_logs.py

```diff
@@ -348,14 +348,7 @@ def __getitem__(self, slice_: slice, /) -> typing.Sequence[AuditLogEntry]:
     def __getitem__(
         self, index_or_slice: typing.Union[int, slice], /
     ) -> typing.Union[AuditLogEntry, typing.Sequence[AuditLogEntry]]:
-        if isinstance(index_or_slice, slice):
-            return tuple(
-                itertools.islice(self.entries.values(), index_or_slice.start, index_or_slice.stop, index_or_slice.step)
-            )
-        elif isinstance(index_or_slice, int):
-            return next(iter(itertools.islice(self.entries.values(), index_or_slice, None)))
-        else:
-            raise TypeError(f"sequence indices must be integers or slices, not {type(index_or_slice).__name__}")
+        return mapping.get_index_or_slice(self.entries, index_or_slice)
```

Comment:
Just noticed the docstring on line 329 says the entries mapping is a sequence. Can we fix this? Ik it is outside the scope of this PR.